sematic/tests/test_user_settings.py

```diff
@@ -1,40 +1,450 @@
 # Standard Library
-import tempfile
-from unittest.mock import PropertyMock, patch
+import os
 
 # Third-party
 import pytest
-import yaml
 
 # Sematic
-import sematic.user_settings
-from sematic.user_settings import SettingsVar, get_all_user_settings, get_user_settings
+import sematic.user_settings as user_settings
+from sematic.api.tests.fixtures import mock_settings_file  # noqa: F401
+from sematic.user_settings import (
+    MissingSettingsError,
+    SettingsVar,
+    UserSettings,
+    _load_settings,
+    delete_profile,
+    delete_user_settings,
+    get_active_user_settings,
+    get_default_settings,
+    get_profiles,
+    get_user_settings,
+    set_active_profile,
+    set_user_settings,
+)
 
+_TEST_FILE = "sematic/tests/data/test_settings.yaml"
+_TEST_DEPRECATED_FILE = "sematic/tests/data/test_version_0_settings.yaml"
 
-@pytest.fixture(scope="function")
-def settings_file():
-    with tempfile.NamedTemporaryFile() as tf:
-        with patch(
-            "sematic.user_settings._settings_file",
-            return_value=tf.name,
-            new_callable=PropertyMock,
-        ):
-            current_settings = sematic.user_settings._settings
-            sematic.user_settings._settings = None
 
-            yield tf
+@pytest.mark.parametrize("mock_settings_file", [_TEST_DEPRECATED_FILE], indirect=True)
+def test_settings_migration(mock_settings_file):  # noqa: F811
+    assert get_active_user_settings() == {SettingsVar.AWS_S3_BUCKET: "sematic-dev"}
+    assert user_settings._settings == UserSettings(
+        active_profile="dev",
+        profiles={"dev": {SettingsVar.AWS_S3_BUCKET.value: "sematic-dev"}},
+    )
 
-            sematic.user_settings._settings = current_settings
+    settings_file_path = user_settings._get_settings_file()
+    # the archive will be created in the bazel workspace, not in the user home
+    assert os.path.exists(f"{settings_file_path}.v0")
 
 
-def test_get_empty_settings(settings_file):
-    assert get_all_user_settings() == {}
+@pytest.mark.parametrize("mock_settings_file", [_TEST_FILE], indirect=True)
+def test_get_active_user_settings(mock_settings_file):  # noqa: F811
+    assert get_active_user_settings() == {SettingsVar.SNOWFLAKE_USER: "foobar"}
 
 
-def test_get_settings(settings_file):
-    settings = {"default": {"SNOWFLAKE_USER": "foobar"}}
-    yaml_output = yaml.dump(settings, Dumper=yaml.Dumper)
-    settings_file.write(bytes(yaml_output, encoding="utf-8"))
-    settings_file.flush()
-    assert get_all_user_settings() == settings["default"]
+@pytest.mark.parametrize("mock_settings_file", [_TEST_FILE], indirect=True)
+def test_get_active_user_settings_env_override(mock_settings_file):  # noqa: F811
+
+    os.environ[str(SettingsVar.SNOWFLAKE_USER.value)] = "baz"
```

Comment:
nit: [this fixture](https://github.com/sematic-ai/sematic/blob/main/sematic/tests/fixtures.py#L49) might make this test cleaner

Reply:
Sweet! Thank you! Done.